### ISSUE TITLE
Is there a way to command it in console?

### ISSUE THREAD

**Issue by blangero** [MaartenBaert/ssr#169]
It would be useful if it can be configured and controlled in console only, to work as a daemon like thing, is there already a solution?

**Comment by TimotheeJeannin** [comment]
@MaartenBaert This would be awesome to be able to run ssr from the command line ! Thank you for working on this ! :)

**Comment by StripedMonkey** [comment]
Currently no, you can start it from the console but you cannot run it
inside the terminal That is on his to-do list if I remember.

**Comment by jokester** [comment]
Hello, thanks for this decent work.

I happen to need a CLI interface, so I want to give a try.
Are there any existing work on this?

Or, can I have some hint like what code to look first (there must be a page explaining how GUI and backend cooperate, cannot find it right now).

**Comment by rodrigofarias77** [comment]
@jokester,

ssr is a much more robust alternative to ffmpeg recording abilities. However, while the CLI is not ready, I am able to sucessfully record screen with ```ffmpeg -vsync drop``` (version 2.5.4), even under heavy load.

Another alternative would be to automate ssr with something like xdotool, but it is a bit tricky...

**Comment by MaartenBaert** [comment]
The main problem is that the record-schedule branch is in a broken state right now because I tried to do too many things at once. And I don't have time to complete the changes that I was trying to make, because something more urgent has popped up (the new GLInject protocol).

If you really need a CLI *now*, then you could probably make one by looking at PageRecord and imitating what it is doing. The GUI and backend are already separate, but PageRecord is currently handling a lot of bookkeeping that should ideally be moved to the backend as well. If you want to make a CLI based on the current master branch, you would have to copy-paste a lot of code from PageRecord (basically everything in the functions StartPage, StopPage, StartOutput, StopOutput, StartInput and StopInput, and probably a few others as well).

In the record-schedule branch, I was working on moving this code to the backend to avoid code duplication when I finally make the CLI. If you make a CLI based on the master branch, I will not be able to merge it, since it would interfere with those changes. But you could do it as a crude temporary solution.

**Comment by jb55** [comment]
It's been awhile... has anyone taken a stab at this yet?

**Comment by RylanSchaeffer** [comment]
Bump? Any updates on this?

**Comment by MaartenBaert** [comment]
Not really, it may not sound like a huge feature but this would require significant changes to the way SSR works, due to the way the recording engine interacts with the GUI. I have other projects that are higher priority right now.
